pkg/skaffold/schema/versions.go

```diff
@@ -237,3 +209,97 @@ func ParseConfigAndUpgrade(filename, toVersion string) ([]util.VersionedConfig,
 	}
 	return upgraded, nil
 }
+
+// configFactoryFromAPIVersion checks that all configs in the input stream have the same API version, and returns a function to create a config with that API version.
+func configFactoryFromAPIVersion(buf []byte) (func() util.VersionedConfig, error) {
+	// This is to quickly check that it's possibly a skaffold.yaml,
+	// without parsing the whole file.
+	if !bytes.Contains(buf, []byte("apiVersion")) {
```

Comment:
i think before doing this, we might want to also check if the file contains `kind: Config` as a further safeguard against users accidentally providing k8s configs to skaffold

Reply:
while that makes sense, I don't want to add that as a new validation in this PR. This is just a refactor of existing code.

Reply:
got it, thought you added this.